### PR TITLE
Export Azure Fake Cloud for reusing

### LINKED ISSUE #88298: Create Azure Disk Fake operations for testing
/assign @ZeroMagic
/kind test
/priority important-soon
/sig cloud-provider
/area provider/azure

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
Via the exported GetTestCloud(), we can reuse the code for the unit tests in Azure related CSI drivers' repo.

**Which issue(s) this PR fixes**:

Fixes #88298 

**Special notes for your reviewer**:
NONE

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```
/sig cloud-provider
/area provider/azure

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_fakes.go [hunk 1]
```diff
@@ -991,3 +998,54 @@ func (f *fakeVMSet) GetDataDisks(nodeName types.NodeName, crt azcache.AzureCache
 func (f *fakeVMSet) GetPowerStatusByNodeName(name string) (string, error) {
 	return "", fmt.Errorf("unimplemented")
 }
+
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_fakes.go]: maybe could add a comment here: why make this func public

**Comment by andyzhangx** [comment]
```
/b/f/w/staging/src/k8s.io/legacy-cloud-providers/azure/azure_test.go:38:2: imported and not used: "k8s.io/kubernetes/vendor/k8s.io/client-go/tools/record"
/b/f/w/staging/src/k8s.io/legacy-cloud-providers/azure/azure_test.go:40:2: imported and not used: "k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure/auth"
compilepkg: error running subcommand: exit status 2
```

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_fakes.go (hunk 1 above)
Comment (ZeroMagic) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_fakes.go]: done

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**